### PR TITLE
3927 check all ssl required

### LINKED ISSUE #3927: Using HTTPS some HTTP calls are still made that will block the browsing if HTTP is not available
In a cartodb installation that only exposes HTTPS interface to the world some requests are still made through HTTP when they shouldn't.

Two points of failure have been identified, but there can be more.

* When you connect to: https://\<yourusername\>.cartodb.com
There is a request to http://\<yourusername\>.cartodb.com that responds with a 302 to https://\<yourusername\>.cartodb.com/dashboard or login.

* If you go to https://\<orgadmin\>.cartodb.com/u/\<orgadmin\>/organization you end up in http://\<orgadmin\>.cartodb.com/u/\<orgadmin\>/organization
<hr>
Branch `3927-check-all-ssl-required` ([PR](https://github.com/CartoDB/cartodb/pull/4011))

### PR DESCRIPTION
**Pull request by Kartones**
Fixes #3927

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4261/))

**Comment by Kartones** [comment]
@juanignaciosl CR please. It is just moving some non-actions to private area and a general SSL-required/allowed check of all controllers as we were missing a few ones (and it screws up if you only allow https traffic)

File: app/controllers/api/json/permissions_controller.rb [hunk 1]
```diff
@@ -1,8 +1,6 @@
 class Api::Json::PermissionsController < Api::ApplicationController
 
-  if Rails.env.production? || Rails.env.staging?
-    ssl_required :show, :update
-  end
+  ssl_required :update if Rails.env.production? || Rails.env.staging?
```
Comment (juanignaciosl) [inline app/controllers/api/json/permissions_controller.rb]: :show doesn't require ssl?

**Comment by Kartones** [comment]
:show doesn't exists anymore :dancer:

**Comment by juanignaciosl** [comment]
:+1: then

**Pull request merged**